File: #1233.remove-sub-folders-from-the-filesystem.cpp
class Solution {
public:
    vector<string> removeSubfolders(vector<string>& folder) {

        vector<string> results;
        std::sort(folder.begin(), folder.end());
        for (string f: folder) {
            if (results.empty()
                || results.back() != f.substr(0, results.back().length())
                || f[results.back().length()] != '/') {
                results.push_back(f);
            } 
        }
        return results;
    }
};

File: #1233.remove-sub-folders-from-the-filesystem.ts
function removeSubfolders(folder: string[]): string[] {
    if (folder.length < 2) {
        return folder;
    }
    
    folder.sort();

    const result: string[] = [];
    let i = 0, j = 1;
    while (i < folder.length) {
        if (j >= folder.length) {
            result.push(folder[i]);
            break;
        }

        if (folder[j].startsWith(folder[i]) && folder[j][folder[i].length] === "/") {
            j++;
        } else {
            result.push(folder[i]);
            i = j;
            j++;
        }
    }
    return result;
};
